### PR TITLE
Add support for chunked httpclient and httpresponse.

### PR DESCRIPTION
**Pull request by claudiamurialdo**
Issue:103034
For .NET framework move SendHeaders() to the begining of the GxWebProcedure to support chunked responses at http procedures.

### CODE REVIEW

File: dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs [hunk 1]
```diff
@@ -115,18 +116,20 @@ protected override void sendCacheHeaders()
...
-				fileName = outputFileName;
-			}
-			if (!string.IsNullOrEmpty(outputType))
-			{
-				fileType = outputType.ToLower();
+				string fileName = GetType().Name;
+				string fileType = "pdf";
+				if (!string.IsNullOrEmpty(outputFileName))
+				{
+					fileName = outputFileName;
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs:125]: This code is about PDF Procedures. Maybe it mixed up from another branch?

File: dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs [hunk 2]
```diff
@@ -138,6 +141,7 @@ protected bool initPrinter(String output, int gxXPage, int gxYPage, String iniFi
 			string idiom;
 			if (!Config.GetValueOf("LANGUAGE", out idiom))
 				idiom = "eng";
+			fileContentInline = true;
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs:144]: This code is about PDF Procedures. Maybe it mixed up from another branch?

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs [hunk 3]
```diff
@@ -1363,6 +1384,25 @@ private Encoding ExtractEncodingFromCharset(string responseText, string regExpP,
 
 		public override string ToString()
 		{
+			if (_chunkedResponse && _receiveStream != null)
```
Comment (ggallotti) [inline dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs:1387]: Shouldn't we ask if the Response Header "Transfer-Encoding"  == chunked before trying to read as chunked? Even if the Developer told GeneXus that chunkedResponseHandling is enabled?

Maybe _receiveStream  is null in that case, if that the case, ignore this comment.

File: dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs (hunk 1 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs]: You are right. Moved this to #826.
I found the error about sending an unnecessary content-disposition header while testing this feature, but it is a separate issue.

File: dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs (hunk 2 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Model/GXWebProcedure.cs]: You are right. Moved this to https://github.com/genexuslabs/DotNetClasses/pull/826.

File: dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs (hunk 3 above)
Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs]: You are correct. The Transfer-Encoding header determines whether the response is chunked (_chunkedResponse is true).

Comment (claudiamurialdo) [reply dotnet/src/dotnetframework/GxClasses/Domain/GxHttpClient.cs]: _chunkedResponse now is true when response.Headers.TransferEncodingChunked.HasValue & response.Headers.TransferEncodingChunked.Value is true

**Pull request merged**